core/src/main/java/org/infinispan/persistence/sifs/LogAppender.java

```diff
@@ -159,7 +159,7 @@ static WriteOperation fromLogRequest(LogRequest logRequest) {
    private void handleRequestCompletion(LogRequest request) {
       int offset = request.getSerializedValue() == null ? ~request.getFileOffset() : request.getFileOffset();
       temporaryTable.set(request.getSement(), request.getKey(), request.getFile(), offset);
-      IndexRequest indexRequest = IndexRequest.update(request.getSement(), request.getKey(), raw(request.getSerializedKey()),
+      IndexRequest indexRequest = IndexRequest.update(request.getSement(), request.getKey(), request.getSerializedKey(),
```

Comment:
nit the method `private byte[] raw(ByteBuffer buffer)` (line 289) can be removed.